src/LogWindow.cpp

```diff
@@ -0,0 +1,229 @@
+/**********************************************************************
+
+Audacity: A Digital Audio Editor
+
+LogWindow.cpp
+
+Paul Licameli split from AudacityLogger.cpp
+
+**********************************************************************/
+#include "LogWindow.h"
+
+#include <wx/filedlg.h>
+#include <wx/frame.h>
+#include <wx/icon.h>
+#include <wx/settings.h>
+#include <wx/textctrl.h>
+#include <wx/weakref.h>
+
+#include "AudacityLogger.h"
+#include "widgets/AudacityMessageBox.h"
+#include "FileNames.h"
+#include "Internat.h"
+#include "MemoryX.h"
+#include "Prefs.h"
+#include "ShuttleGui.h"
+
+#include "../images/AudacityLogoAlpha.xpm"
+
+// If wxLogWindow is used and initialized before the Mac's "root" window, then
+//     Audacity may crash when terminating.  It's not fully understood why this occurs
+//     but it probably has to do with the order of deletion.  However, deferring the
+//     creation of the log window until it is actually shown circumvents the problem.
+enum
+{
+   LoggerID_Save = wxID_HIGHEST + 1,
+   LoggerID_Clear,
+   LoggerID_Close
+};
+
+namespace {
+Destroy_ptr<wxFrame> sFrame;
+wxWeakRef<wxTextCtrl> sText;
+
+struct LogWindowListener : public PrefsListener
+{
+   // PrefsListener implementation
+   void UpdatePrefs() override;
+};
+// Unique PrefsListener can't be statically constructed before the application
+// object initializes, so use Optional
+Optional<LogWindowListener> pListener;
```

Comment:
Also I can't find place where using `UpdatePrefs()`, I see that pointer initializing below by `.emplace()` method, but don't see where it use.

Reply:
I am already find, this is callback when `audacity` preferences updated.